### ISSUE TITLE
Deprecate findin(a, b) in favor or find(occursin(b), a)

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#24967]
As part of the unification of search and find functions (https://github.com/JuliaLang/julia/issues/10593), we should deprecate `findin(a, b)` in favor of the more explicit `find(occursin(b), a)`, which is equivalent to `find(x -> x in b, a)`. `occursin` is already implemented in https://github.com/JuliaLang/julia/pull/24673.

**Comment by gabrielgellner** [comment]
oh man `occursin` is hard to parse for me, I just naturally break `occur_sin` visually. Bases hatred of underscores can be rough. The logic of this on the other hand is super sweet.

**Comment by StefanKarpinski** [comment]
We could have a single-argument `in(v)` that does what `occursin(v)` does – it reads naturally, and I'm not sure what else it could possibly mean.

**Comment by rfourquet** [comment]
> We could have a single-argument in(v)

:100: It seems very natural to me, and FWIW, I introduced `_in` to do just that in a [PR](https://github.com/JuliaLang/julia/pull/23528/files#diff-d39084ec0fef2ad65e4b5fe0fcfaab8eR104)

**Comment by nalimilan** [comment]
That's tempting, but I'd rather have a general plan regarding currying functions, as it could be really confusing to have that (reverse) currying function if we don't do that more systematically. For example, we now have `equalto(x)`, which could be replaced with `==(x)` in the same vein.

@stevengj also [proposed](https://github.com/JuliaLang/julia/issues/10593#issuecomment-337288327) `find(_ in c, x)`, pointing at https://github.com/JuliaLang/julia/issues/5571.

**Comment by stevengj** [comment]
See #24990 for an implementation of `find(_ in c, x)` etcetera.

**Comment by nalimilan** [comment]
Actually, something like `occursin` is needed so that we can dispatch to more efficient methods than the default linear search. We can't do that with custom anonymous functions.

I've added a commit replacing `findin(a, b)` with `find(occursin(b), a)` to https://github.com/JuliaLang/julia/pull/24673.

**Comment by JeffBezanson** [comment]
Fixed by #24673.
